compiler/parser/chapel.ypp

```diff
@@ -704,8 +708,29 @@ if_stmt:
 ;
 
 try_stmt:
-  TTRY     stmt { $$ = TryStmt::build(false, $2); }
-| TTRYBANG stmt { $$ = TryStmt::build(true,  $2); }
+  TTRY     expr            TSEMI         { $$ = TryStmt::build(false, $2); }
+| TTRYBANG expr            TSEMI         { $$ = TryStmt::build(true,  $2); }
+| TTRY     assignment_stmt               { $$ = TryStmt::build(false, $2); }
+| TTRYBANG assignment_stmt               { $$ = TryStmt::build(true,  $2); }
+| TTRY     block_stmt      catch_stmt_ls { $$ = TryStmt::build(false, $2, $3); }
+| TTRYBANG block_stmt      catch_stmt_ls { $$ = TryStmt::build(true,  $2, $3); }
+;
+
+catch_stmt_ls:
+                           { $$ = buildChapelStmt(); }
+| catch_stmt_ls catch_stmt { $1->insertAtTail($2); }
+;
+
+catch_stmt:
+  TCATCH catch_expr block_stmt { $$ = CatchStmt::build($2, $3); }
+;
+
+
+/* TODO: would be nice to parenthesize a catch */
```

Comment:
@vasslitvinov - actually only the no-parens version parses right now; the TODO is to allow it with parens as I showed above.
``` chapel
try {
  ...
} catch e:Error {
  ...
}
```

parses now.